### DISCUSSION TITLE
Setting User Rating and Last Position Heard

### DISCUSSION THREAD

**Discussion by Mbucari** [mkb79/audible-cli#134] [Q&A] [answered]
Do you know how to send/set new last position heard and user ratings to Audible? Is there an API endpoint for those?

**Comment by mkb79** [comment, marked as answer]
The endpoint for the user rating is `1.0/catalog/review`. You can add an user rating with a `PUT` request. 

Here is an example request body for adding a review

```json
{
  "review" : {
    "asin" : "B08M5FQ2CP",
    "rating_dimensions" : [
      {
        "dimension" : "performance",
        "rating" : 4
      },
      {
        "dimension" : "overall",
        "rating" : 4
      },
      {
        "dimension" : "story",
        "rating" : 4
      }
    ]
  }
}
```

Keep in mind that a user rating must be verified by Audible to be shown. This can take some minutes. 

The url for the last position heared is `https://cde-ta-g7g.amazon.com/FionaCDEServiceEngine/sidecar`. 
You can set a new position with a `POST` request using the url above. But you must add query params to the url. 

Example for a complete url is `https://cde-ta-g7g.amazon.com/FionaCDEServiceEngine/sidecar?countryCode=DE&device_lto=60&type=AUDI`

The body must be xml encoded and looks like

```xml
<annotations version="1.0" timestamp="2022-12-21T07:03:02+0100">
    <book key="B08M5FQ2CP" type="AUDI" version="23555421" guid="CR!JXYV7SZVQN381CD856CFJXHGHHPM:23555421" format="AAX_44_128"><last_heard action="modify" begin="7727567" timestamp="2022-12-21T07:03:02+0100" /></book>
</annotations>
```

Best regards

**Reply by Mbucari** [reply]
Thanks!
